applications/ParticleMechanicsApplication/custom_conditions/particle_based_conditions/mpm_particle_penalty_dirichlet_condition.cpp

```diff
@@ -205,20 +211,53 @@ void MPMParticlePenaltyDirichletCondition::CalculateAll(
             rRightHandSideVector *= m_penalty * this->GetIntegrationWeight();
         }
     }
-    else{
-        // To improve stability: use identity matrix to avoid nonzero diagonal LHS matrix
-        if ( CalculateStiffnessMatrixFlag == true )
-        {
-            noalias(rLeftHandSideMatrix) = IdentityMatrix(matrix_size);
-        }
-    }
```

Comment:
Just to understand... why can this be removed?

Reply:
Hi @ncrescenzio 
this can be removed as the builder and solver automatically fill one at the main diagonal of the stiffness matrix if the row is zero.
For contact this is especially important, as we want to deactivate the boundary condition imposition. However if we set the identity matrix this will introduce an error.